teleprobe/src/client.rs

```diff
@@ -104,13 +107,42 @@ impl ElfMetadata {
         if let Some(section) = obj_file.section_by_name(".teleprobe.timeout") {
             let data = section.data()?;
             if data.len() == 4 {
-                meta.timeout = Some(u32::from_le_bytes(data.try_into().unwrap()) as u64)
+                timeout = Some(u32::from_le_bytes(data.try_into().unwrap()) as u64)
             } else {
                 warn!(".teleprobe.timeout contents are not a valid u32.")
             }
         }
 
-        Ok(meta)
+        let mut hasher = Blake2b::new(32)?;
+        for section in &mut obj_file.sections() {
+            let section_name = match section.name() {
+                Ok(name) => name,
+                _ => continue,
+            };
+
+            if section_name == "" || section_name.starts_with(".debug_") || section_name.starts_with(".teleprobe.") {
```

Comment:
I would hash the `.teleprobe.*` sections, they can influence whether test succeeds or fails.